particle-iot/device-os
# Mesh network scan fixes

### Problem

This PR introduces the following changes:

- Make the duration of an active scan configurable.
- Remove duplicate network entries from the network scan results.
- Remove our workarounds for OpenThread bugs that were fixed in the upstream repo and pulled into our dependency branch (see https://github.com/openthread/openthread/pull/3246).

### Steps to Test

- Set up a mesh network with at least 2 nodes (more is better).
- Perform a few network scans using the CLI and a device which is not a member of the same network:
```
particle mesh scan my_device
```
- Make sure there are no duplicate network entries in the scan results.

### References

- [ch31161]
